report.md

```diff
@@ -2,6 +2,24 @@
 
 ## Subsystem 1 (MFA)
 
+Multifactor authentication will apply to all users and will aim for a balance between confidentiality and availability.
+
+### First Factor
+The first step of authentication will involve a username and password. The username will be stored as a hash. The password will be stored as a hash with salt. The salt for each user will be unique and will be generated upon registration. The salt will be stored as plain text.
+
+### Second Factor
+The second factor will be triggered on the completion of the first factor. On passing the first stage a random string will be generated by the server and stored temporarily in the database with the associated user in hash form. An sms will be sent to the user based on the registered phone number with the digits. The user will be redirected to another form to key in the digits. He will have 2 minutes to do so after which the login process will fail and the temporary string will be replaced with nonces. On success the user will receive a session and be able to login and the string will be replaced with nonces.
+
+### Third factor for special actions
+The third factor will be applicable only to administrators. The administrators will have a specially assigned BLE token with a private symmetric??? key. When logging in as an administrator, the token will have to be present and scanned before gaining access to the account on top of the first and second factor.
+
+### Services and tools to be used
+GraphQL and PHP for the initial registration. PHP crypto libraries will be used to encrypt the usernames,passwords and other data.

```

Comment:
possible to do away with PHP and use Java crypto instead?

Reply:
Building on our previous discussions in the previous [PR](https://github.com/IFS4205-2018-Sem1-Team1/design-report/pull/8#discussion_r215240550):

We can use [Spring & GraphQL](https://stackoverflow.com/questions/48050509/graphql-spring-boot-upload-binary) to perform file upload. So we don't need to use PHP for file upload.

Both [Spring Security](https://www.baeldung.com/spring-security-registration-password-encoding-bcrypt) and [Java](https://stackoverflow.com/questions/1205135/how-to-encrypt-string-in-java) can encrypt data.

@gjlam95 do you think we should use Spring Security, or just plain Java?

Reply:
i think plain java is fine leh. Later use spring security then kena question why we dont use for authentication :/

Reply:
Honestly it doesn't matter what we use. Spring is very likely to be Javascript or PHP crypto because that is was Facebook uses. Anyway, the fact is all these crypto libraries follow the same formula. We just pick one and stick with it should be ok. 

Any word on the symmetric???? @gngster